esp8266/scripts/neopixel.py

```diff
@@ -19,10 +20,19 @@ def _reorder(self, val):
             yield val[max(i, self.bpp - 1)]
 
     def __setitem__(self, index, val):
+        if hasattr(index, 'start'):
```

Comment:
I can use `isinstance(index, slice)` here, as I would in normal Python, because there is no `slice` type available.